railties/lib/rails/engine/commands.rb

```diff
@@ -1,12 +1,7 @@
-require "rails/command"
+unless defined?(APP_PATH)
+  if File.exists?(File.expand_path("test/dummy/config/application.rb", ENGINE_ROOT))
+    Object.const_set(:APP_PATH, File.expand_path("test/dummy/config/application", ENGINE_ROOT))
```

Comment:
Why do we have to go through `const_set`?

Won't a better place for this be somewhere the `Rails::Command` can set this internally?